### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by Bornittah**
### Create models
-  Create model classes for all entities as shown in the [ERD diagram](../images/blog_app_erd_v1_1.png).
-  Set up associations between models.
    - Remember that `author_id` column in `posts` table should be the foreign_key for the `users` table.

### Use models to insert data
- Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- Open rails console for your project.
- Create at least one user by running the following code:
    ```ruby
       first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
       second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
    ```
- Create at least 4 posts written by one of the users you created by running the following code:
    ```ruby
       first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
    ```
- Create at least 6 posts comments for one of the posts you created by running the following code:
    ```ruby
       Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
    ```
- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.

### Create custom methods
Your models should include some additional methods.
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,3 +1,10 @@
 class Post < ApplicationRecord
-  belongs_to :author_id
+  belongs_to :author, class_name: :User, foreign_key: :author_id
+  has_many :comments, foreign_key: :post_id
+  has_many :likes, foreign_key: :post_id
+  after_save :update_posts_counter
+
```
Comment (fobadara) [inline app/models/post.rb:6]: [REQUIRED] It is a requirement for post model to have a method that returns the 5 most recent comments for a given post.

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
 class Comment < ApplicationRecord
-  belongs_to :author_id
-  belongs_to :post_id
+  belongs_to :author, class_name: :User, foreign_key: :author_id
+  belongs_to :post, foreign_key: :post_id
```
Comment (fobadara) [inline app/models/comment.rb:3]: [REQUIRED] Kindly include a method that updates the comment counter for a post because this is a requirement

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @Bornittah, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Linters are passing

## Required Changes ♻️
_Check the comments under the review._
I had to include some comments here because it could not be done inline

[REQUIRED] It is a requirement to add relationships to your user model and to create the 3 most recent post methods. Excluding this will cause issues when your app is run.
[REQUIRED] Kindly remove the `live Demo` section in your README since you have no `live demo` link


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @Bornittah, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which haven't been addressed yet.


 ## Highlights
✅ Linter is passing

 ## Required Changes ♻️
_Check the comments under the review._
**IMPORTANT**
I think you missed this in the last review but there was no way for me to add  inline comments since you didn't make changes to the affected files in this PR
**[REQUIRED] It is a requirement to add relationships to your user model and to create the 3 most recent post methods. Excluding this will cause issues when your app is run.
[REQUIRED] Kindly remove the live Demo section in your README since you have no live demo link**

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @Bornittah, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did a great job making the changes requested by the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**